Lib/test/test_typing.py

```diff
@@ -3355,6 +3355,15 @@ def barfoo4(x: BA3): ...
             {"x": typing.Annotated[int | float, "const"]}
         )
 
+    def test_get_type_hints_annotated_in_union(self):  # bpo-46603
+        def with_union(x: int | list[Annotated[str, 'meta']]): ...
```

Comment:
Are all the other paths covered already? This test doesn't seem to cover the _GenericAlias branch in _strip_annotations.

Reply:
Yes, other branches are reported to be covered by `test_typing` and/or `test_types` somewhere.